strapi/strapi
# enhancement: restrict config filenames

### What does it do?

Creates a blacklist of config files that will be ignored by the config file loader and ignores them

### Why is it needed?

Strapi loads all files found in the config dir into strapi.config. However, we need to start restricting filenames to case-insensitive unique, alphanumeric characters, that do not conflict with Strapi environment variables that start with `STRAPI_{ReservedPortion}` so that in the future we can safely load environment variables into config and know how to map the values in them.

Also restricts filenames that could have overridden root-level config values such as 'launchedAt', although in a later refactoring those will be moved to a better location.

As a bonus, also adds a warning for mistaken filenames like plugin.js (instead of plugins.js)

### How to test it?

Try creating config files with a reserved filename prefix, a duplicate filename (ie, ADMIN.js and admin.js) and a mistaken filename, and see that they are not loaded and a warning is logged instead.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request
